catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,748 @@
+                QueryRequestImpl.class);
+        verify(mockSortedQueryMonitorFactory).createMonitor(any(CompletionService.class),
+                any(Map.class),
+                responseArgumentCaptor.capture(),
+                requestArgumentCaptor.capture(),
+                any(List.class));
+
+        assertThat(federateResponse.getRequest()
+                .getQuery()
+                .getStartIndex(), is(strategy.DEFAULT_MAX_START_INDEX + 5));
+        // Modified offset is set to 1
+        assertThat(requestArgumentCaptor.getValue()
+                .getQuery()
+                .getStartIndex(), is(1));
+    }
+
+    @Test
+    public void testOffsetResultHandler() throws Exception {
+        QueryResponseImpl originalResults = mock(QueryResponseImpl.class), offsetResultQueue = mock(
+                QueryResponseImpl.class);
+        int pageSize = 1, offset = 1;
+
+        Result mockResult = mock(Result.class);
+        MetacardImpl mockMetacard = mock(MetacardImpl.class);
+
+        List<Result> results = new ArrayList<>();
+
+        CachingFederationStrategy.OffsetResultHandler offsetResultHandler =
+                new CachingFederationStrategy.OffsetResultHandler(originalResults,
+                        offsetResultQueue,
+                        pageSize,
+                        offset);
+
+        when(originalResults.hasMoreResults()).thenReturn(true);
+        when(mockResult.getMetacard()).thenReturn(mockMetacard);
+        when(mockMetacard.getId()).thenReturn("mock metacard");
+        when(originalResults.take()).thenReturn(mockResult);
+
+        doAnswer(invocation -> results.add(mockResult)).when(offsetResultQueue)
+                .addResult(any(Result.class), any(Boolean.class));
+
+        offsetResultHandler.run();
+
+        assertThat(results.size(), is(1));
+        assertThat(results.get(0)
+                .getMetacard()
+                .getId(), is("mock metacard"));
+        verify(offsetResultQueue, atLeastOnce()).addResult(any(Result.class), any(Boolean.class));
+    }
+
+    @Test
+    public void testCatchPluginExecutionException() throws Exception {
+        PreFederatedQueryPlugin mockPlug = mock(PreFederatedQueryPlugin.class);
+        when(mockPlug.process(any(Source.class),
+                any(QueryRequest.class))).thenThrow(new PluginExecutionException());
+        strategy = new CachingFederationStrategy(queryExecutor,
+                Arrays.asList(mockPlug),
+                new ArrayList<>(),
+                cache,
+                cacheExecutor,
+                mock(ValidationQueryFactory.class),
+                new CacheQueryFactory(new GeotoolsFilterBuilder()));
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        strategy.federate(Arrays.asList(mock(Source.class)), fedQueryRequest);
+        verify(mockPlug).process(any(Source.class), any(QueryRequest.class));
```

Comment:
❗️ We should do like we did for the other test and have two mocks. Make sure the second one is actually called in this case.